src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -144,6 +139,53 @@ internal abstract partial class AbstractAddParameterCheckCodeRefactoringProvider
 
         protected abstract bool CanOffer(SyntaxNode body);
 
+        private async Task<Document> UpdateDocumetForRefactoring(
+            Document document,
+            SyntaxNode functionDeclaration,
+            IBlockOperation blockStatementOpt,
+            List<int> listOfParametersOrdinals,
+            int position,
+            CancellationToken cancellationToken)
+        {
+
+            foreach (var index in listOfParametersOrdinals)
+            {
+                // Updates functionDeclaration and uses it to get the first valid ParameterNode using the ordinals (index).
+                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var token = root.FindToken(position);
+                var firstparameterNode = GetParameterNode(token, position);
+                functionDeclaration = firstparameterNode.FirstAncestorOrSelf<SyntaxNode>(IsFunctionDeclaration);
+
+                var currentNode = GetParameterNodeAtIndex(functionDeclaration, index);
+                if (currentNode == null)
+                {
+                    continue;
+                }
+
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var parameter = (IParameterSymbol)semanticModel.GetDeclaredSymbol(currentNode, cancellationToken);
+                var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+
+                if (!CanOfferRefactoring(functionDeclaration, semanticModel, syntaxFacts, cancellationToken, out blockStatementOpt))
+                {
+                    continue;
+                }
+
+                // If parameter is a string, default check would be IsNullOrEmpty - updates document
+                if (parameter.Type.SpecialType == SpecialType.System_String)
+                {
+                    document = await AddStringCheckAsync(document, parameter, functionDeclaration, (IMethodSymbol)parameter.ContainingSymbol, blockStatementOpt, nameof(string.IsNullOrEmpty), cancellationToken).ConfigureAwait(false);
+                    continue;
+                }
+
+                // For all other parameters, add null check - updates document
+                document = await AddNullCheckAsync(document, parameter, functionDeclaration,
+                    (IMethodSymbol)parameter.ContainingSymbol, blockStatementOpt, cancellationToken).ConfigureAwait(false);
```

Comment:
this appears to be a loop, that's updating the document for each and every parameter we add a null check for.  that doesn't seem desirable.  can we not add all the null checks at once, and produce the final document?

Reply:
I tried to minimize the amount of change for this PR. I can do what you have suggested in the followup PR.

Reply:
> I tried to minimize the amount of change for this PR.

this would still be contained to effectively just this method.  we just need to reimplement this in a saner way :)

Reply:
or, alternatively, when is this 'followup PR' happening?  will this PR go into a feature branch?